src/screens/stationSelect.tsx

```diff
@@ -76,13 +77,16 @@ const StationSelectScreen: React.FC<StationSelectProps> = ({ navigation }) => {
         arriveStationCrsCode: arriveStationCrsCode,
       });
     } else {
-      // TODO stretch goals: add some error messaging here
+      setShowErrorMessage(true);
+      setErrorMessage('The departure and arrival stations cannot be the same.');
+      return;
     }
+    setShowErrorMessage(false);
```

Comment:
This should probably be in your if statement before navigating elsewhere to make the 'logic tree' a bit clearer.